vulcanai2/models/utils.py

```diff
@@ -0,0 +1,62 @@
+from collections import OrderedDict
+import torch
+from torch.autograd import Variable
+import torch.nn as nn
+import json
+ 
+ 
+def get_size(summary_dict, output):
+    if isinstance(output, tuple):
+        for i in xrange(len(output)):
+            summary_dict[i] = OrderedDict()
+            summary_dict[i] = get_size(summary_dict[i],output[i])
+    else:
+        summary_dict['output_shape'] = list(output.size())
+    return summary_dict
+ 
+def get_output_shapes(input_size, model):
```

Comment:
I would like to see more documentation here since it seems quite complicated to extract the output shapes of each layer. Also if a function is in a class, we shouldn't need to pass it a model since it should just look at `self`. it doesn't make sense to use it as a static class when we've already instantiated it in the script that it is made it.

Reply:
Tbh it was quite complicate for me to understand as well. 
ref links: https://pytorch.org/tutorials/beginner/former_torchies/nn_tutorial.html#forward-and-backward-function-hooks
https://github.com/sksq96/pytorch-summary/blob/master/torchsummary/torchsummary.py

Reply:
ok for now let's put a link to this resource in the code and modify the function signature to just use the internal model instead of having to pass it one

Reply:
so in that case it has to be moved to the `BaseNetwork`, right?

Reply:
yeah good idea. it will be used in all networks